agent/src/softwarecontaineragent.h

```diff
@@ -226,7 +228,7 @@ class SoftwareContainerAgent : protected softwarecontainer::JSONParser
     // List of containers in use
     std::map<ContainerID, SoftwareContainerPtr> m_containers;
     // List of pre-loaded containers
-    std::vector<SoftwareContainerPtr> m_preloadedContainers;
+    std::vector<std::pair<ContainerID, SoftwareContainerPtr>> m_preloadedContainers;
```

Comment:
I'm guessing there is a good reason for having a vector of pairs instead of a map, but I don't see it.